arm-software/ci/automerge.py

```diff
@@ -79,15 +80,21 @@ def prefix_current_commit_message(git_repo: Git) -> None:
     git_repo.run_cmd(["commit", "--amend", "--message=" + commit_msg])
 
 
-def merge_commit(git_repo: Git, to_branch: str, commit_hash: str, ignored_paths: list[str], dry_run: bool) -> None:
+def merge_commit(git_repo: Git, to_branch: str, commit_hash: str, ignored_paths: list[str], dry_run: bool, verbose: bool) -> None:
     logger.info("Merging commit %s into %s", commit_hash, to_branch)
     git_repo.run_cmd(["switch", to_branch])
+    if verbose:
+        current_head = git_repo.run_cmd(["log", "--no-walk", "HEAD", "--pretty=reference"])
+        logger.debug("Current HEAD of %s is %s", to_branch, current_head)
     git_repo.run_cmd(["merge", commit_hash, "--no-commit", "--no-ff"], check=False)
     restore_changes_to_ignored_files(git_repo, ignored_paths)
     if has_unresolved_conflicts(git_repo):
         logger.info("Merge failed")
         git_repo.run_cmd(["merge", "--abort"])
         raise MergeConflictError(commit_hash)
+    if verbose:
+        commit_content = git_repo.run_cmd(["diff", "--cached"])
+        logger.debug("Changes being commited:\n%s", commit_content)
```

Comment:
I think this might be a step too far. Some commits will have _very_ large diffs, e.g. ones which add a huge number of tests, or churn a lot of existing ones. I think printing all the commit ids is enough for the things we're currently anticipating trying to debug.

Reply:
That's a good point. I removed it for now.